napalm_nxos/nxos.py

```diff
@@ -327,27 +522,75 @@ def commit_config(self):
             raise ReplaceConfigException('No config loaded.')
 
     def _delete_file(self, filename):
-        self.device.show('terminal dont-ask', raw_text=True)
-        self.device.show('delete {}'.format(filename), raw_text=True)
-        self.device.show('no terminal dont-ask', raw_text=True)
+        commands = [
+            'terminal dont-ask',
+            'delete {}'.format(filename),
+            'no terminal dont-ask'
+        ]
+        for command in commands:
+            self._send_command(command, fmt='text')
 
     def discard_config(self):
         if self.loaded:
             self.merge_candidate = ''  # clear the buffer
         if self.loaded and self.replace:
             try:
-                self._delete_file(self.fc.dst)
+                self._delete_file(self.replace_file)
             except CLIError:
                 pass
         self.loaded = False
 
+    def _rollback_ssh(self, backup_file):
+        result = self.device.send_command('rollback running-config file %s' % backup_file)
+        if 'completed' not in result.lower():
+            raise ReplaceConfigException(result)
+        self._save_ssh()
+
     def rollback(self):
         if self.changed:
-            self.device.rollback(self.backup_file)
-            self.device.save()
+            if self.ssh_connection:
+                self._rollback_ssh(self.backup_file)
+            else:
+                self.device.rollback(self.backup_file)
+                self.device.save()
             self.changed = False
 
-    def get_facts(self):
+    def _apply_key_map(self, key_map, table):
+        new_dict = {}
+        for key, value in table.items():
+            new_key = key_map.get(key)
+            if new_key:
+                new_dict[new_key] = str(value)
+        return new_dict
+
+    def _convert_uptime_to_seconds(self, uptime_facts):
+        seconds = int(uptime_facts['up_days']) * 24 * 60 * 60
+        seconds += int(uptime_facts['up_hours']) * 60 * 60
+        seconds += int(uptime_facts['up_mins']) * 60
+        seconds += int(uptime_facts['up_secs'])
+        return seconds
+
+    def _get_facts_ssh(self):
+        final_facts = {}
+        show_ver_output = json.loads(self.device.send_command_timing('show version | json'))
```

Comment:
`send_command` is probably better to use here (as there should be trailing prompt in the output).